src/rc/seedrng.c

```diff
@@ -371,18 +399,31 @@ static int seed_from_file_if_exists(const char *filename, bool credit, struct bl
 
 static void populate_global_paths(void)
 {
-	SEED_DIR = getenv("SEEDRNG_SEED_DIR");
-	if (!SEED_DIR || !*SEED_DIR)
-		SEED_DIR = "/var/lib/seedrng";
-	LOCK_FILE = getenv("SEEDRNG_LOCK_FILE");
-	if (!LOCK_FILE || !*LOCK_FILE)
-		LOCK_FILE = "/var/run/seedrng.lock";
-	xasprintf(&CREDITABLE_SEED, "%s/seed.credit", SEED_DIR);
-	xasprintf(&NON_CREDITABLE_SEED, "%s/seed.no-credit", SEED_DIR);
+	if (!lock_file) {
+		lock_file = xstrdup(getenv("SEEDRNG_LOCK_FILE"));
+		if (!lock_file || !*lock_file)
+			xasprintf(&lock_file, "%s", "/var/run/seedrng.lock");
```

Comment:
Here, why do you use "%s", when you could use strdup on the string literal? But more importantly, why allocate at all? Just keep the type as `const char *`, and assign either `getenv()` to it, or the string constant to it.